mobile/lib/modules/backup/providers/backup.provider.dart

```diff
@@ -314,7 +314,7 @@ class BackupNotifier extends StateNotifier<BackUpState> {
 
       // Perform Backup
       state = state.copyWith(cancelToken: CancellationToken());
-      _backupService.backupAsset(
+      await _backupService.backupAsset(
```

Comment:
is this to allow multiple assets to be uploaded at the same time?

Reply:
for error handling, I had the most strange debugging issues (and crashes) without all async functions returning a Future and awaiting other async functions

Reply:
Is it only when running as a background service?

Reply:
Have the issues only when running as background service. The change would of course affect both. I have just re-checked if this `await` is still necessary for running as a service: Yes. If I remove it, running `startBackupProcess` fails/stops the executing thread

Reply:
finally figured out what's causing the different behavior (background service vs in foreground app):
The background process is started by the Android WorkManager and executes the new Java/Kotlin `BackupWorker`. This starts the Dart/Flutter runtime in a new thread. I Had to make the worker aware of running another async thread by not immediately stopping after finishing the code on the Java/Kotlin side. Instead a `ResolvableFuture` is used to tell the Android system that the process completed (successfully or failed). This signal is send by the dart/flutter thread once it finishes executing (i.e. `return Future.value(true);` in `callbackDispatcher` method handler). If the executed dart code calls an `async` function without `await` this function is executed in yet another thread and the main dart code continues, finished, and in turn signals Android that the background work is finished. The Android system then kills the entire process. Yet, this non-main async dart function (e.g. `backupAsset`) was still running and is killed along with its process.

tl;dr: all `async` functions need to be awaited else the thread/process is killed while executing them in the background.

Not await an async function is also a problem in the foreground app: If an error occurs in that function, it is silently ignored!